react/more-react-concepts/refs_and_memoization/refs_and_memoization.md

```diff
@@ -0,0 +1,384 @@
+### Introduction
+
+How can we do DOM manipulations that we usually do in vanilla JavaScript? Is it even possible in React because of its declarative nature? How about memoization? As we all know, when a state has been changed, React will try to re-render the component, which means that it will destroy all local variables not controlled by React and re-execute them. This mental model also works for a regular function that you create, as a component is just a function. Since this is the case, wouldn't it slow down the app if we have a very expensive calculation re-executed with each re-render?
+
+### Lesson Overview
+
+This section contains a general overview of topics you will learn in this lesson.
+
+- Explore `useRef` hook and its use cases
+- Explain memoization and how `useCallback` and `useMemo` can be used
+
+### The useRef hook
```

Comment:
```suggestion
### The `useRef` hook
```

Reply:
note: according to the [style guide](https://github.com/TheOdinProject/curriculum/blob/main/LAYOUT_STYLE_GUIDE.md#headings), we don't use code snippets in headers

Reply:
Deleting the sentence-casing issues. I mistook it with the style guide of a different repository.